bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -5324,34 +5334,7 @@ void ReplicaImp::onExecutionFinish() {
   handleDeferredRequests();
   auto seqNumToStopAt = ControlStateManager::instance().getCheckpointToStopAt();
   if (seqNumToStopAt.value_or(0) == lastExecutedSeqNum) ControlStateManager::instance().wedge();
-  if (seqNumToStopAt.has_value() && seqNumToStopAt.value() > lastExecutedSeqNum && isCurrentPrimary()) {
-    // If after execution, we discover that we need to wedge at some future point, push a noop command to the incoming
-    // messages queue.
-    LOG_INFO(GL, "sending noop command to bring the system into wedge checkpoint");
-    concord::messages::ReconfigurationRequest req;
-    req.sender = config_.replicaId;
-    req.command = concord::messages::WedgeCommand{config_.replicaId, true};
-    // Mark this request as an internal one
-    std::vector<uint8_t> data_vec;
-    concord::messages::serialize(data_vec, req);
-    req.signature.resize(SigManager::instance()->getMySigLength());
-    SigManager::instance()->sign(data_vec.data(), data_vec.size(), req.signature.data());
-    data_vec.clear();
-    concord::messages::serialize(data_vec, req);
-    std::string strMsg(data_vec.begin(), data_vec.end());
-    auto requestSeqNum =
-        std::chrono::duration_cast<std::chrono::microseconds>(getMonotonicTime().time_since_epoch()).count();
-    auto crm = new ClientRequestMsg(internalBFTClient_->getClientId(),
-                                    RECONFIG_FLAG,
-                                    requestSeqNum,
-                                    strMsg.size(),
-                                    strMsg.c_str(),
-                                    60000,
-                                    "wedge-noop-command-" + std::to_string(lastExecutedSeqNum + 1));
-    // Now, try to send a new PrePrepare message immediately, without waiting to a new batch
-    onMessage(std::unique_ptr<ClientRequestMsg>(crm));
-    tryToSendPrePrepareMsg(false);
-  }
+  primaryPushNoOpIfWedgePending(lastExecutedSeqNum + 1);
```

Comment:
shouldn't it be without the +1?

Reply:
idk, just extracted a function for huge duplication